src/App.tsx

```diff
@@ -8,7 +8,58 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+
+import { Todo } from './types/Todo';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [initialTodos, setInitialTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [query, setQuery] = useState('');
+  const [statusSelect, setStatusSelect] = useState('all');
+
+  const onSelectedTodo = (todoId: number) => {
+    setSelectedTodo(todos.find(todo => todo.id === todoId) || null);
+  };
+
+  const setClose = (item: null) => {
+    setSelectedTodo(item);
+  };
+
+  const searchByInput = (todoTitle: string, searchInput: string) => {
+    return todoTitle.toLowerCase().includes(searchInput.toLowerCase());
+  };
+
+  useEffect(() => {
+    getTodos().then(todo => {
+      setTodos(todo);
+      setInitialTodos(todo);
+      setIsLoading(false);
+    });
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code